### PR TITLE
demos: Make alpn identifiers work with mbedtls

### PR DESCRIPTION
**Pull request by paulbartell**
* Move AWS IoT Core alpn identifiers from demo_config.h to
  mqtt_agent_task.c.
* Replace openssl formatted ALPN identifier strings with mbedtls
  compatible ALPN identifier strings.

### CODE REVIEW

**Review by ravibhagavandas — DISMISSED** [review]

File: examples/common/mqtt_agent/mqtt_agent_task.c [hunk 1]
```diff
@@ -180,6 +180,23 @@
  */
 #define mqttexampleEVENT_BITS_ALL    ( ( EventBits_t ) ( ( 1ULL << MQTT_AGENT_NUM_STATES ) - 1U ) )
 
+
```
Comment (n9wxu) [inline examples/common/mqtt_agent/mqtt_agent_task.c:183]: Please consolidate the ALPN choices and the decision for which ALPN to use in a single place.

**Review by n9wxu — CHANGES_REQUESTED** [review]

File: examples/common/mqtt_agent/mqtt_agent_task.c (hunk 1 above)
Comment (paulbartell) [reply examples/common/mqtt_agent/mqtt_agent_task.c]: Adjusted as requested. Please let me know what you think.

File: examples/common/mqtt_agent/mqtt_agent_task.c [hunk 2]
```diff
@@ -506,26 +506,51 @@ static MQTTStatus_t prvCreateMQTTConnection( bool xIsReconnect )
...
-     * second entry must remain NULL. */
-    const char * pcAlpnProtocols[] = { NULL, NULL };
+    /*
+     * When connecting AWS IoT Core via mqtt on port 443, an alpn is required.
+     * For more information, refer to the AWS IoT Core Device documentation at:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/protocols.html
+     */
+    if( ulBrokerPort == 443U )
+    {
+#if defined( democonfigCLIENT_USERNAME )
```
Comment (ravibhagavandas) [inline examples/common/mqtt_agent/mqtt_agent_task.c:524]: Would it be more user friendly to define a config such as `democonfigUSE_CUSTOM_AUTH` ?
`democonfigCLIENT_USERNAME` could also be used for other  use cases like metrics reporting.

Comment (paulbartell) [reply examples/common/mqtt_agent/mqtt_agent_task.c:524]: democonfigCLIENT_USERNAME is better if targeting non-AWS mqtt services.

**Review by ravibhagavandas — DISMISSED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**